src/Magento/FunctionalTestingFramework/Test/Objects/ActionObject.php

```diff
@@ -496,6 +496,10 @@ private function findAndReplaceReferences($objectHandler, $inputString)
 
             $replacement = $this->resolveParameterization($parameterized, $replacement, $match, $obj);
 
+            if (is_array($replacement)) {
```

Comment:
This check should either live in the elseif block in 497, or in the `resolveEntityDataObjectReference()` function. This conditional only ever applies to EntityDataObjects, so checking it every time is unecessary.